pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,434 @@
+                            onReportButtonClick={this.onReportButtonClick}
+                            problemState={this.state.problemState}
+                            reportLinks={this.state.reportLinks} />;
+    }
+
+    updateStatusFromBus() {
+        get_problem_property(this.props.problem, "Reports")
+                .done((reports) => {
+                    for (const report of reports[0].v) {
+                        if (!("WORKFLOW" in report[1])) {
+                            continue;
+                        }
+                        if (this.props.workflow[0] !== report[1].WORKFLOW.v.v) {
+                            continue;
+                        }
+                        if (report[0] === "ABRT Server" || report[0] === "uReport") {
+                            continue;
+                        }
+                        const reportLinks = [];
+                        if ("URL" in report[1]) {
+                            reportLinks.push(report[1].URL.v.v);
+                        }
+                        this.setState({
+                            problemState: ProblemState.REPORTED,
+                            reportLinks: reportLinks,
+                        });
+                        break;
+                    }
+                });
+        get_problem_property(this.props.problem, "CanBeReported")
+                .done((reportable) => {
+                    if (!reportable) {
+                        this.setState({ problemState: ProblemState.UNREPORTABLE });
+                    }
+                });
+    }
+}
+
+function WorkflowRow(props) {
+    let status = props.message;
+
+    if (props.problemState === ProblemState.REPORTED) {
+        if (props.reportLinks.length === 1) {
+            status = (
+                <a href={props.reportLinks[0]} rel="noopener noreferrer" target="_blank">
+                    View report <i className="fa fa-external-link" aria-hidden="true" />
+                </a>
+            );
+        } else if (props.reportLinks.length > 1) {
+            const reportLinks = props.reportLinks.map((reportLink, index) => [
+                index > 0 && ", ",
+                <a href={reportLink} key={index.toString()} rel="noopener noreferrer" target="_blank">
+                    {index + 1} <i className="fa fa-external-link" aria-hidden="true" />
+                </a>
+            ]);
+            status = <p>Reports: {reportLinks}</p>;
+        } else {
+            status = "Reported; no links available";
+        }
+    }
+
+    const cancelButton = (
+        <button className="btn btn-danger" onClick={props.onCancelButtonClick}>
+            Cancel
+        </button>
+    );
+    const reportButton = (
+        <button className="btn btn-primary"
+                disabled={props.problemState !== ProblemState.REPORTABLE}
+                onClick={props.problemState === ProblemState.REPORTABLE && props.onReportButtonClick}>
```

Comment:
If you used to conditionally omit it with onClick={condition && value}, pass onClick={condition ? value : undefined} instead.

Reply:
minimally to keep react from complaining :D 
But I guess they have some reason why this check exists

Reply:
Again, treat me like I’m stupid. Complaining about what?

Reply:
I have a feeling that these are the warnings you mentioned, but I just don’t see them popping up.

Reply:
if you run `autogen.sh`  with `--enable-debug` then you should see that in your browser console.

Reply:
Aha, perfect! I was completely lost about these things.